aurora_module/rds_aurora_cluster.tf

```diff
@@ -0,0 +1,152 @@
+/**
+ * Copyright (c) 2011 - 2019, Coveo Solutions Inc.
+ */
+
+locals {
+  instances_count = "${var.engine_mode == "serverless" ? 0 : var.instances_count}"
+  snapshot_identifier_found = "${lookup(var.optional_parameters, "snapshot_identifier", "") == "" ? false : true}"
+  internal_security_groups_vpc_ids = ["${aws_security_group.rds_cluster_security_group.id}"]
+  security_groups_vpc_ids = "${concat(local.internal_security_groups_vpc_ids, var.vpc_security_group_ids)}"
+}
+
+resource "aws_db_subnet_group" "db_subnet_group" {
+  subnet_ids = "${var.subnet_ids}"
+
+  name        = "${lookup(var.optional_parameters, "db_subnet_group_name", "${var.custom_identifier}")}"
+  description = "${lookup(var.optional_parameters, "subnet_group_description", "${var.custom_identifier}")}"
+  tags        = "${var.subnet_group_tags}"
+}
+
+resource "aws_ssm_parameter" "rds_cluster_master_username" {
+  count = "${var.store_master_creds_to_ssm ? 1 : 0}"
+  name  = "${lookup(var.optional_parameters, "parameter_store_path", "${var.custom_identifier}")}/${lookup(var.optional_parameters, "parameter_store_username_key", "Username")}"
+  type  = "String"
+  value = "${var.master_username}"
+}
+
+resource "aws_ssm_parameter" "rds_cluster_master_password" {
+  count = "${var.store_master_creds_to_ssm ? 1 : 0}"
+  name  = "${lookup(var.optional_parameters, "parameter_store_path", "${var.custom_identifier}")}/${lookup(var.optional_parameters, "parameter_store_password_key", "Password")}"
+  type  = "SecureString"
+  value = "${var.master_password}"
+
+  key_id = "${lookup(var.optional_parameters, "master_pass_kms_key_id", "")}"
+}
+
+resource "aws_secretsmanager_secret" "master_user" {
+  count = "${var.use_secretmanager ? 1 : 0}"
+  name                = "${lookup(var.optional_parameters, "secret_manager_rotation_name_master_user", "")}"
+  rotation_lambda_arn = "${lookup(var.optional_parameters, "secret_manager_rotation_lambda_arn_master_user", "")}"
+
+  kms_key_id = "${lookup(var.optional_parameters, "secret_kms_key_id", "")}"
+  policy = "${lookup(var.optional_parameters, "secret_policy", "")}"
+  recovery_window_in_days = "${lookup(var.optional_parameters, "secret_recovery_window_in_days", "7")}"
+
+  rotation_rules {
+    automatically_after_days = "${lookup(var.optional_parameters, "secret_automatically_after_days", 7)}"
+  }
+}
+
+resource "aws_secretsmanager_secret" "master_password" {
```

Comment:
With the AWS Secrets manager, the username and the password must be in the same secret, otherwise, the rotation Lambda cannot proceed to the rotation.

For example, if you follow [this article](https://aws.amazon.com/blogs/security/rotate-amazon-rds-database-credentials-automatically-with-aws-secrets-manager/), the secret generated by the AWS Secrets manager UI is:
```JSON
{
  "username": "test_jpl",
  "password": "some_super_password_i_remember",
  "engine": "mysql",
  "host": "my.db-endpoint",
  "port": 3306,
  "dbClusterIdentifier": "my-db-name"
}
```

With AWS Secrets manager, the "secret" contains everything needed by the rotation Lambda function.

Therefore, instead of creating two `aws_secretsmanager_secret` resources, we should create only one that contains everything needed by the rotation Lambda function. 

Make sense? 🙂

Reply:
Right, it makes total sense. 😄